python/python/lance/lance/__init__.pyi

```diff
@@ -99,4 +124,356 @@ class LanceBlobFile:
+        fragment_id: Optional[int],
+        reader: ReaderLike,
+        **kwargs,
+    ): ...
+    def id(self) -> int: ...
+    def metadata(self) -> _FragmentMetadata: ...
+    def count_rows(self, _filter: Optional[str] = None) -> int: ...
+    def take(
+        self,
+        row_indices: List[int],
+        columns: Optional[Union[List[str], Dict[str, str]]],
+    ) -> pa.RecordBatch: ...
+    def scanner(
+        self,
+        columns: Optional[List[str]],
+        columns_with_transform: Optional[List[Tuple[str, str]]],
+        batch_size: Optional[int],
+        filter: Optional[str],
+        limit: Optional[int],
+        offset: Optional[int],
+        with_row_id: Optional[bool],
+        batch_readahead: Optional[int],
+        **kwargs,
+    ) -> _Scanner: ...
+    def add_columns_from_reader(
+        self,
+        reader: ReaderLike,
+        batch_size: Optional[int],
+    ) -> Tuple[_FragmentMetadata, LanceSchema]: ...
+    def add_columns(
+        self,
+        transforms: Dict[str, str] | BatchUDF | ReaderLike,
+        read_columns: Optional[List[str]],
+        batch_size: Optional[int],
+    ) -> Tuple[_FragmentMetadata, LanceSchema]: ...
+    def delete(self, predicate: str) -> Optional[_Fragment]: ...
+    def schema(self) -> pa.Schema: ...
+    def data_files(self) -> List[_DataFile]: ...
+    def deletion_file(self) -> Optional[str]: ...
+    @property
+    def physical_rows(self) -> int: ...
+    @property
+    def num_deletions(self) -> int: ...
+
+class _DataFile:
+    def path(self) -> str: ...
+    def field_ids(self) -> List[int]: ...
+    def __lt__(self, other: BFloat16) -> bool: ...
+    def __le__(self, other: BFloat16) -> bool: ...
+    def __eq__(self, other: object) -> bool: ...
+    def __ne__(self, other: object) -> bool: ...
+    def __gt__(self, other: BFloat16) -> bool: ...
+    def __ge__(self, other: BFloat16) -> bool: ...
+
+class _FragmentMetadata:
+    @staticmethod
+    def from_json(payload: str) -> _FragmentMetadata: ...
+    def json(self) -> str: ...
+    def data_files(self) -> Iterable[str]: ...
+    def deletion_file(self): ...
+    @property
+    def physical_rows(self) -> int: ...
+    @property
+    def num_deletions(self) -> int: ...
+    @property
+    def num_rows(self) -> int: ...
+    @property
+    def id(self) -> int: ...
+
+def _write_dataset(
+    reader: pa.RecordBatchReader, uri: str | Path | _Dataset, params: Dict[str, Any]
+): ...
+def _write_fragments(
+    dataset_uri: str | Path | _Dataset,
+    reader: ReaderLike,
+    mode: str,
+    max_rows_per_file: int,
+    max_rows_per_group: int,
+    max_bytes_per_file: int,
+    progress: Optional[FragmentWriteProgress],
+    data_storage_version=Optional[str],
+    storage_options=Optional[Dict[str, str]],
+): ...
+def _json_to_schema(schema_json: str) -> pa.Schema: ...
+def _schema_to_json(schema: pa.Schema) -> str: ...
+def trace_to_chrome(file: Optional[str] = None, level: Optional[str] = None): ...
```

Comment:
Shouldn't this be in the tracing package?

Reply:
do you mean it should be organized like optimize.pyi, debug.pyi? But I get atttribute access issue, when I use pyright to check these files. putting all definition in __init__.pyi can solve this issue, because actually there's only one dynlib.

Reply:
I reorg the file path, now pyright doesn't complain error